### PR TITLE
contrib: fix package manager & dependencies management

### PR DESCRIPTION
**Pull request by 0xcryptovenom**
Currently the project contains a lockfile both for `yarn` v1 (`yarn.lock`)  & for `npm` (`package-lock.json`).

The README mentions both the use of `yarn` & of `npx`.

Yarnv1 has been obsolete for years now.

[pnpm](https://pnpm.io/) is by far the most popular, the best package manager nowadays, but npm is the simplicity's choice, and much better than it used to be when it was out-hyped by yarn.

The runtime & development dependencies defined in the `package.json` were mixed, with pretty much everything in `dependencies`.

Since the Adrena frontend is a web application built using Next.js & using a bundler, as opposed to, ie, a Backend Node.js application (which it still is, to some extent), things will still work if we decide to put all the dependencies in one or the other.

However, for the sake of dependencies management, crucial for security, performance, DX, maintanability, we correctly categorise the dependencies:
- in `dependencies`, list the dependencies which are actually used at runtime, on the server or on the client, even if those end up being bundled by Webpack or whaterver bundler used by Next.js
- in `devDependencies`, list the dependencies which are only relevant for the development workflow, the build process, and irrelevant in production, at runtime.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,59 +12,63 @@
...
+    "eslint-plugin-simple-import-sort": "^10.0.0",
+    "husky": "^8.0.3",
+    "postcss": "^8.4.21",
+    "prettier": "^2.8.7",
+    "sass": "^1.59.3",
+    "typescript": "4.9.5"
+  },
+  "overrides": {
+    "jito-ts": {
+      "@solana/web3.js": "$@solana/web3.js"
```
Comment (0xcryptovenom) [inline package.json:67]: Force `jito-ts` which is a "deep" transitive dependency to use our version of `@solana/web3.js`.

File: package.json [hunk 2]
```diff
@@ -12,59 +12,63 @@
...
+    }
+  },
+  "engines": {
+    "node": "20.18.0",
```
Comment (0xcryptovenom) [inline package.json:71]: To match with the highest we can get on Vercel

File: package.json [hunk 3]
```diff
@@ -12,59 +12,63 @@
...
+    "npm": "10.8.2"
```
Comment (0xcryptovenom) [inline package.json:72]: same

**Comment by adrena-orex** [comment]
Thanks for the contribution, have rebased and created a new PR that will be merged to dev and main soon

https://github.com/AdrenaFoundation/frontend/pull/284

**Pull request closed without merging**